### PR TITLE
Increase usage of ScreenCoordsXY on viewport functions and structs

### PR DESCRIPTION
**Pull request by tupaschoal**
I changed `rct_viewport` and `rct_window` to use `ScreenCoordsXY`, not sure if this is a problem, as they used to have comments on the size and paddings

### CODE REVIEW

File: src/openrct2-ui/windows/Banner.cpp [hunk 1]
```diff
@@ -133,10 +133,9 @@ rct_window* window_banner_open(rct_windownumber number)
     window_init_scroll_widgets(w);
 
     auto banner = GetBanner(w->number);
-    int32_t view_x = banner->position.x << 5;
-    int32_t view_y = banner->position.y << 5;
+    auto bannerViewPos = banner->position.ToCoordsXY();
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Banner.cpp]:
```suggestion
    auto bannerViewPos = banner->position.ToCoordsXY().ToTileStart();
```
and remove the += at line 154.

File: src/openrct2-ui/windows/Banner.cpp [hunk 2]
```diff
@@ -347,15 +346,14 @@ static void window_banner_viewport_rotate(rct_window* w)
 
     auto banner = GetBanner(w->number);
 
-    int32_t view_x = (banner->position.x << 5) + 16;
-    int32_t view_y = (banner->position.y << 5) + 16;
-    int32_t view_z = w->frame_no;
+    auto bannerViewPos = CoordsXYZ{ banner->position.ToCoordsXY() + CoordsXY{ 16, 16 }, w->frame_no };
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Banner.cpp]:
```suggestion
    auto bannerViewPos = CoordsXYZ{ banner->position.ToCoordsXY().ToTileStart(), w->frame_no };
```

File: src/openrct2-ui/windows/Player.cpp [hunk 3]
```diff
@@ -581,7 +580,7 @@ static void window_player_set_page(rct_window* w, int32_t page)
     {
         if (w->viewport == nullptr)
         {
-            viewport_create(w, w->x, w->y, w->width, w->height, 0, 128 * 32, 128 * 32, 0, 1, SPRITE_INDEX_NULL);
+            viewport_create(w, { w->x, w->y }, w->width, w->height, 0, { 128 * 32, 128 * 32, 0 }, 1, SPRITE_INDEX_NULL);
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Player.cpp]:
```suggestion
            viewport_create(w, { w->x, w->y }, w->width, w->height, 0, TileCoordsXYZ(128, 128, 0).ToCoordsXYZ, 1, SPRITE_INDEX_NULL);
```

File: src/openrct2-ui/windows/Sign.cpp [hunk 4]
```diff
@@ -156,10 +156,9 @@ rct_window* window_sign_open(rct_windownumber number)
     window_init_scroll_widgets(w);
 
     auto banner = GetBanner(w->number);
-    int32_t view_x = banner->position.x << 5;
-    int32_t view_y = banner->position.y << 5;
+    auto signViewPos = banner->position.ToCoordsXY();
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Sign.cpp]: Use ToTileStart()

File: src/openrct2-ui/windows/Sign.cpp [hunk 5]
```diff
@@ -365,15 +364,14 @@ static void window_sign_viewport_rotate(rct_window* w)
 
     auto banner = GetBanner(w->number);
 
-    int32_t view_x = (banner->position.x << 5) + 16;
-    int32_t view_y = (banner->position.y << 5) + 16;
-    int32_t view_z = w->frame_no;
+    auto signViewPos = CoordsXYZ{ banner->position.ToCoordsXY() + CoordsXY{ 16, 16 }, w->frame_no };
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Sign.cpp]: ditto

File: src/openrct2-ui/windows/Sign.cpp [hunk 6]
```diff
@@ -433,14 +430,14 @@ rct_window* window_sign_small_open(rct_windownumber number)
     w->var_492 = tile_element->AsWall()->GetSecondaryColour();
     w->var_48C = tile_element->AsWall()->GetEntryIndex();
 
-    view_x += 16;
-    view_y += 16;
+    signViewPos += CoordsXY{ 16, 16 };
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Sign.cpp]: ditto

File: src/openrct2-ui/windows/Viewport.cpp [hunk 7]
```diff
@@ -94,13 +94,13 @@ rct_window* window_viewport_open()
     w->number = _viewportNumber++;
 
     // Create viewport
-    viewport_create(w, w->x, w->y, w->width, w->height, 0, 128 * 32, 128 * 32, 0, 1, SPRITE_INDEX_NULL);
+    viewport_create(w, { w->x, w->y }, w->width, w->height, 0, { 128 * 32, 128 * 32, 0 }, 1, SPRITE_INDEX_NULL);
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Viewport.cpp]: See other remark about using proper conversions.

**Comment by Gymnasiast** [comment]
I think the comments on the size and paddings are from the era where we were still calling RCT2 functions.

I think the PR is good, but I did find some bits you need to address.

File: src/openrct2-ui/windows/Banner.cpp (hunk 1 above)
Comment (tupaschoal) [reply src/openrct2-ui/windows/Banner.cpp]: Can't this theoretically return a different `TileElement*`? Because the sum happens after the `map_get_first_element_at`

File: src/openrct2-ui/windows/Banner.cpp (hunk 2 above)
Comment (tupaschoal) [reply src/openrct2-ui/windows/Banner.cpp]: Did you mean `ToTileCentre`?

I remembered about this function, but I was unsure if flooring would be alright.

File: src/openrct2-ui/windows/Banner.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2-ui/windows/Banner.cpp]: No. Both represent the same tile, and the structure holding them divides the x and y by 32 anyway.

File: src/openrct2-ui/interface/Window.cpp [hunk 8]
```diff
@@ -133,8 +133,6 @@ rct_window* window_create(
 
     // Setup window
     w->classification = cls;
-    w->var_4B8 = -1;
```
Comment (Gymnasiast) [inline src/openrct2-ui/interface/Window.cpp]: Why was this removed?

File: src/openrct2-ui/windows/GameBottomToolbar.cpp [hunk 9]
```diff
@@ -421,8 +421,8 @@ static void window_game_bottom_toolbar_draw_left_panel(rct_drawpixelinfo* dpi, r
     if (!(gParkFlags & PARK_FLAGS_NO_MONEY))
     {
         rct_widget widget = window_game_bottom_toolbar_widgets[WIDX_MONEY];
-        int32_t x = w->x + (widget.left + widget.right) / 2;
-        int32_t y = w->y + (widget.top + widget.bottom) / 2 - (line_height == 10 ? 5 : 6);
+        int32_t x = w->windowPos.x + (widget.left + widget.right) / 2;
+        int32_t y = w->windowPos.y + (widget.top + widget.bottom) / 2 - (line_height == 10 ? 5 : 6);
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/GameBottomToolbar.cpp]: Note to self or @AaronVanGeffen : shouldn't we just divide `line_height` by 2 here?

File: src/openrct2-ui/interface/Window.cpp (hunk 8 above)
Comment (tupaschoal) [reply src/openrct2-ui/interface/Window.cpp]: No one was reading from it.

File: src/openrct2-ui/windows/GameBottomToolbar.cpp (hunk 9 above)
Comment (AaronVanGeffen) [reply src/openrct2-ui/windows/GameBottomToolbar.cpp]: No, some fonts use a line height of 14 or 15 while the glyphs are offset to compensate. It's a little messy, but the main reason why I hardcoded this.

File: src/openrct2/interface/Viewport.cpp [hunk 10]
```diff
@@ -1,4 +1,4 @@
-﻿/*****************************************************************************
+/*****************************************************************************
```
Comment (Gymnasiast) [inline src/openrct2/interface/Viewport.cpp]: :bomb:
